src/client/eppo-client-experiment-container.spec.ts

```diff
@@ -0,0 +1,85 @@
+import { MOCK_UFC_RESPONSE_FILE, readMockUFCResponse } from '../../test/testHelpers';
+import * as applicationLogger from '../application-logger';
+import { MemoryOnlyConfigurationStore } from '../configuration-store/memory.store';
+import { Flag, ObfuscatedFlag } from '../interfaces';
+
+import EppoClient, { IFlagExperiment } from './eppo-client';
+import { initConfiguration } from './test-utils';
+
+type Container = { name: string };
+
+describe('getExperimentContainer', () => {
+  global.fetch = jest.fn(() => {
+    const ufc = readMockUFCResponse(MOCK_UFC_RESPONSE_FILE);
+    return Promise.resolve({
+      ok: true,
+      status: 200,
+      json: () => Promise.resolve(ufc),
+    });
+  }) as jest.Mock;
+
+  const controlContainer: Container = { name: 'Control Container' };
+  const treatment1Container: Container = { name: 'Treatment Variation 1 Container' };
+  const treatment2Container: Container = { name: 'Treatment Variation 2 Container' };
+  const treatment3Container: Container = { name: 'Treatment Variation 3 Container' };
+
+  let client: EppoClient;
+  let flagExperiment: IFlagExperiment<Container>;
+  let getStringAssignmentSpy: jest.SpyInstance;
+  let loggerWarnSpy: jest.SpyInstance;
+
+  beforeEach(async () => {
+    const storage = new MemoryOnlyConfigurationStore<Flag | ObfuscatedFlag>();
+    await initConfiguration(storage);
+    client = new EppoClient(storage);
+    client.setIsGracefulFailureMode(true);
+    flagExperiment = {
```

Comment:
I generally like to leave anything with Arrays and objects that are referenced by the production code in a before each to prevent against potential mutations that could affect subsequent tests that are difficult to diagnose. It's unlikely, but it can happen.

Reply:
If you feel strongly though, I can move it.

Reply:
Nope that's defensible too!